src/h5web/visualizations/heatmap/Tooltip.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { ReactElement, useCallback } from 'react';
+import { Dom, PointerEvent, useThree } from 'react-three-fiber';
+import { TooltipWithBounds, useTooltip } from '@vx/tooltip';
+import { scaleLinear } from 'd3-scale';
+import { format } from 'd3-format';
+import styles from './HeatmapVis.module.css';
+
+interface Props {
+  data: number[][];
+  dims: [number, number];
+}
+
+function Tooltip(props: Props): ReactElement {
+  const { data, dims } = props;
+  const [rows, cols] = dims;
+
+  const { camera, size, intersect } = useThree();
+  const { width, height } = size;
+
+  const {
+    tooltipOpen,
+    tooltipTop,
+    tooltipLeft,
+    tooltipData,
+    showTooltip,
+    hideTooltip,
+  } = useTooltip<[number, number]>();
+
+  // Scales to compute data coordinates from unprojected mesh coordinates
+  const xCoordScale = scaleLinear()
+    .domain([-height / 2, height / 2])
+    .range([0, rows]);
+  const yCoordScale = scaleLinear()
+    .domain([-width / 2, width / 2])
+    .range([0, cols]);
+
+  // Update tooltip when pointer moves
+  // When panning, events are handled and stopped by texture mesh and do not reach this mesh (which is behind)
+  const onPointerMove = useCallback(
+    (evt: PointerEvent) => {
+      const { zoom } = camera;
+      const projectedPoint = camera.worldToLocal(evt.unprojectedPoint.clone());
+
+      const xCoord = Math.floor(xCoordScale(evt.unprojectedPoint.y));
+      const yCoord = Math.floor(yCoordScale(evt.unprojectedPoint.x));
```

Comment:
```suggestion
      const yCoord = Math.floor(yCoordScale(evt.unprojectedPoint.y));
```
and the other way round above ;)

Reply:
Arf, silly me 🤦‍♂️